Microsoft.Azure.Cosmos/src/Headers/CosmosQueryResponseMessageHeaders.cs

```
@@ -13,15 +14,23 @@ namespace Microsoft.Azure.Cosmos
     internal class CosmosQueryResponseMessageHeaders : Headers
     {
         public CosmosQueryResponseMessageHeaders(
+            double requestCharge,
+            string activityId,
+            SubStatusCodes subStatusCode,
             string continauationToken,
             string disallowContinuationTokenMessage,
             ResourceType resourceType,
-            string containerRid)
+            string containerRid,
+            int itemCount)
         {
+            this.RequestCharge = requestCharge;
```

what's required or not required ?

---

The headers are not documented for all scenarios.  For example if there is a networking error and the request never makes it to the service then there will be no request charge. Causing a argument exception is the code for these cases will be extremely painful for users as it will hide the real exception.